### PR TITLE
Refactor Modal component.

### PR DESCRIPTION
**Pull request by kostyalmm**
## Summary

Addresses issue #2193

## Relevant technical choices

Refactors Modal component to be functional component.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/components/modal.js [hunk 1]
```diff
@@ -19,28 +19,23 @@
 /**
  * WordPress dependencies
  */
-import { Component, createPortal } from '@wordpress/element';
+import { createPortal, useEffect, useState } from '@wordpress/element';
```
Comment (felixarntz) [inline assets/js/components/modal.js:22]: I think this file was accidentally kept. It should be removed in favor of `Modal.js`.

File: assets/js/components/Modal.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+function Modal( { children } ) {
+	// Using state as we need `el` to not change when the component re-renders
+	const [ el ] = useState( document.createElement( 'div' ) );
+
+	useEffect( () => {
+		const root = document.querySelector( '.googlesitekit-plugin' ) || document.body;
+		root.appendChild( el );
+
+		return () => root.removeChild( el );
+	}, [] );
```
Comment (felixarntz) [inline assets/js/components/Modal.js:33]: This looks good as far as I can tell. @tofumatt Would be great to get your review here as well.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@kostyalmm Pretty much LGTM, just the old file still needs to be removed.

Also, can you add a QA Brief to the issue please?

**Review by eugene-manuilov — APPROVED** [review]

**Review by felixarntz — APPROVED** [review]
Great!

**Pull request merged**